qiskit_ibm_runtime/runtime_job.py

```diff
@@ -196,11 +199,11 @@ def status(self) -> JobStatus:
 
     def in_final_state(self) -> bool:
         """Return whether the job is in a final job state such as ``DONE`` or ``ERROR``."""
-        return self._status in JOB_FINAL_STATES
+        return self.status() in self.JOB_FINAL_STATES
```

Comment:
Would it be better in these cases to reuse `job.in_final_state()`?

Reply:
We can't here because `in_final_state()` calls `status()` which would result in an infinite loop